src/components/course/EnrollButton.jsx

```diff
@@ -56,104 +149,107 @@ export default function EnrollButton() {
   );
 
   const enrollmentUrl = useMemo(
-    () => {
-      if (subscriptionLicense) {
-        const enrollmentFailedParams = { ...qs.parse(location.search) };
-        enrollmentFailedParams[ENROLLMENT_FAILED_QUERY_PARAM] = true;
-        const coursePageUrl = `${process.env.BASE_URL}${location.pathname}`;
-
-        const enrollOptions = {
-          license_uuid: subscriptionLicense.uuid,
-          course_id: key,
-          enterprise_customer_uuid: enterpriseConfig.uuid,
-          next: `${process.env.LMS_BASE_URL}/courses/${key}/course`,
-          // Redirect back to the same page with a failure query param
-          failure_url: `${coursePageUrl}?${qs.stringify(enrollmentFailedParams)}`,
-        };
-        return `${process.env.LMS_BASE_URL}/enterprise/grant_data_sharing_permissions/?${qs.stringify(enrollOptions)}`;
-      }
-
-      // TODO: the "Enroll" button does not yet support other subsidy types beyond subscription
-      // licenses. as such, the enrollment url for codes/offers is unknown at this time.
-      return null;
-    },
-    [subscriptionLicense, enterpriseConfig, key],
+    () => getEnrollmentUrl({
+      catalogList,
+      enterpriseConfig,
+      key,
+      location,
+      offers,
+      offersCount,
+      offersLoading,
+      sku,
+      subscriptionLicense,
+    }),
+    [catalogList, enterpriseConfig, key, location, offers, offersCount, offersLoading, sku, subscriptionLicense],
```

Comment:
my goodness this is a long list of variables that can change, but hey what do we do right?

Reply:
Yeah, for things that are less of a function, this is the point where I split it into a separate component. But this doesn't really deserve a react component, so, yeah, it is what it is; most changes will affect it.